src/commands.ts

```diff
@@ -343,6 +362,48 @@ export class CosmosDBCommands {
             }
         }
     }
+
+    public static async deleteDocDBDocument(doc: DocDBDocumentNode, explorer: CosmosDBExplorer): Promise<void> {
+        if (doc) {
+            const confirmed = await vscode.window.showWarningMessage("Are you sure you want to delete document '" + doc.label + "'?", "Yes", "No");
+            if (confirmed === "Yes") {
+                const masterKey = await doc.coll.db.getPrimaryMasterKey();
+                const endpoint = await doc.coll.db.getEndpoint();
+                const client = new DocumentClient(endpoint, { masterKey: masterKey });
+                const docLink = doc.getDocLink();
+                await new Promise((resolve, reject) => {
+                    client.deleteDocument(docLink, (err) => {
+                        err ? reject(new Error(err.body)) : resolve();
+                    });
+                });
+                explorer.refresh(doc.coll);
+            }
+        }
+    }
+
+    public static async updateDocDBDocument(document: DocDBDocumentNode): Promise<void> {
+        //get the data from the editor
+        const masterKey = await document.coll.db.getPrimaryMasterKey();
+        const endpoint = await document.coll.db.getEndpoint();
+        const client = new DocumentClient(endpoint, { masterKey: masterKey });
+        const editor = vscode.window.activeTextEditor;
+        const newdocument = JSON.parse(editor.document.getText());
+        const docLink = newdocument._self;
+        await new Promise((resolve, reject) => {
+            client.replaceDocument(docLink, newdocument,
+                { accessCondition: { type: 'IfMatch', condition: newdocument._etag } },
+                (err, updated) => {
+                    if (err) {
+                        reject(new Error(err.body));
```

Comment:
I assume this will show a readable message if the accessCondition fails?

Reply:
It says 'Conflict', followed by some JSON like message. 
Talked to Jing offline about formatting this error message, but that work item seems low priority.